### DISCUSSION TITLE
Dynamic Rendering

### DISCUSSION THREAD

**Discussion by foremtehan** [inertiajs/inertia#1093] [Ideas]
Excuse me, Does inertia supports Dynamic rendering ?

> Dynamic rendering is the process of serving content based on the user agent that calls it. This means serving a client-side rendered version of your site for human users and a separate, server-side version for search engines.

Its really great for saving server resources, It only SSR for google bots

**Comment by ajnsn** [comment]
Hey @foremtehan 
If you are using Laravel, [this answer](https://github.com/inertiajs/inertia-laravel/issues/345#issuecomment-1008007642) could be interesting for you. 

You could write your own [HttpGateway](https://github.com/inertiajs/inertia-laravel/blob/master/src/Ssr/HttpGateway.php) and check for the user agent string - something like `Request::server('HTTP_USER_AGENT')` should work. 
If you do not want SSR, just return `null`.
